common/src/main/java/apoc/result/VirtualNode.java

```diff
@@ -67,9 +67,8 @@ public VirtualNode(Node node, List<String> propertyNames, boolean wrapNodeIDs) {
         Objects.requireNonNull(node, ERROR_NODE_NULL);
         final long id = node.getId();
         // if node is already virtual, we return the same id
-        this.id = id < 0 || wrapNodeIDs ? id : -id - 1;
+        this.id = id < 0 || wrapNodeIDs ? id : MIN_ID.decrementAndGet();
         // to not overlap this nodes ids with ids from VirtualNode(Label[] labels, Map<String, Object> props)
-        MIN_ID.updateAndGet(x -> Math.min(x, this.id));
         this.labels.addAll(Util.labelStrings(node));
         String[] keys = propertyNames.toArray(new String[propertyNames.size()]);
         this.props.putAll(node.getProperties(keys));
```

Comment:
Referring to:

```
this.elementId = node.getElementId();
```

just keeping the element id. Is that right?

It looks like a potential source of confusion, particularly, in a scenario like the one mentioned in my other comment. Is it, at least, worthwhile mentioning this behavior in the documentation?

I know this is outside the fix.

Reply:
Yep, I agree, I wanted to keep that separate from this fix though :)